includes/class-admin-plugins-screen.php

```diff
@@ -0,0 +1,234 @@
+<?php
+/**
+ * Newspack Admin Plugins Screen
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Newspack functionality on the WP Admin Plugins screen.
+ */
+class Admin_Plugins_Screen {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_filter( 'all_plugins', [ $this, 'inject_managed_plugins' ] );
+		add_filter( 'plugin_action_links', [ $this, 'modify_action_links' ], 10, 3 );
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_scripts_and_styles' ] );
+		add_action( 'admin_action_newspack_install_plugin', [ $this, 'handle_plugin_install' ] );
+		add_action( 'all_admin_notices', [ $this, 'plugin_install_notices' ] );
+	}
+
+	/**
+	 * Add all the managed plugins to the plugins table.
+	 *
+	 * @param  array $plugins Array of plugin info.
+	 * @return array Modified $plugins.
+	 */
+	public function inject_managed_plugins( $plugins ) {
```

Comment:
Very minor UI nit: if you manually delete a managed plugin, following the ajax response the row will be removed from the table. If you refresh the page it goes back to being displayed as a "managed plugin, uninstalled" but this a scenario where the UI can be left in a mildly incorrect state.

Reply:
Yeah. There isn't a simple way to handle that problem. I can make an issue after this is merged so we don't forget about it, but I think we can live with it for now.

Reply:
Agreed. This is very minor UI complaint, in an area which is already the fallback UI, so if this will take more than 15 min to correct we should just let it go.